dirty_cat/similarity_encoder.py

```diff
@@ -402,6 +402,8 @@ def fit(self, X, y=None):
                 self.vocabulary_count_matrices_.append(vocabulary_count_matrix)
                 self.vocabulary_ngram_counts_.append(vocabulary_ngram_count)
 
+        self.drop_idx_ = self._compute_drop_idx()
```

Comment:
It seems that `drop_idx` and `_compute_drop_idx` were added in sklearn 0.21, so that's why for skleanr 0.20 the tests failed. You can use the code below to check that sklearn version is at least 0.21 and in that case call the function `compute_drop_idx`.

```
if LooseVersion(sklearn_version) > LooseVersion('0.20'):
    self.drop_idx_ = self._compute_drop_idx()
```